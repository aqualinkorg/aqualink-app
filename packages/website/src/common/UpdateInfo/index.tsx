import React from 'react';
import { Theme, Grid, Box, Typography } from '@mui/material';
import makeStyles from '@mui/styles/makeStyles';
import { grey } from '@mui/material/colors';
import UpdateIcon from '@mui/icons-material/Update';
import Chip from '../Chip';

const CHIP_SMALL_DEFAULT_WIDTH = 48;
const CHIP_LARGE_DEFAULT_WIDTH = 60;
const UPDATE_ICON_SIZE = 24;
const UPDATE_ICON_RIGHT_MARGIN = 4;

const UpdateInfo = ({
  relativeTime,
  timeText,
  image,
  imageText,
  live = false,
  frequency,
  href,
  withMargin = false,
  chipWidth,
  subtitle,
  onClick,
}: UpdateInfoProps) => {
  const classes = useStyles({ chipWidth });
  return (
    <Grid
      className={`${classes.updateInfo} ${withMargin && classes.withMargin}`}
      container
      justifyContent="space-between"
      alignItems="center"
      item
    >
      <Grid item className={classes.dateInfoWrapper}>
        <Grid container alignItems="center" justifyContent="center">
          <Grid item>
            <UpdateIcon className={classes.updateIcon} fontSize="small" />
          </Grid>
          <Grid item className={classes.dateInfo}>
            <Box display="flex" flexDirection="column" width="100%">
              <Typography className={classes.updateInfoText} variant="caption">
                {relativeTime
                  ? `${timeText} ${relativeTime}`
                  : 'No data available'}
              </Typography>
              <Typography
                title={subtitle}
                className={classes.updateInfoText}
                variant="caption"
              >
                {frequency ? `Updated ${frequency}` : subtitle}
              </Typography>
            </Box>
          </Grid>
        </Grid>
      </Grid>
      <Grid item style={{ display: 'flex', justifyContent: 'flex-end' }}>
        <Chip
          live={live}
          href={live ? undefined : href}
          image={image}
          imageText={imageText}
          onClick={onClick}
        />
      </Grid>
    </Grid>
  );
};

const useStyles = makeStyles((theme: Theme) => ({
  updateInfo: {
    backgroundColor: grey[400],
    color: grey[700],
    padding: 4,
    minHeight: 40,
    flexWrap: 'nowrap',
  },
  withMargin: {
    marginTop: 32,
  },
  updateIcon: {
    marginRight: UPDATE_ICON_RIGHT_MARGIN,
    height: UPDATE_ICON_SIZE,
    width: UPDATE_ICON_SIZE,
  },
  updateInfoText: {
    width: '99%',
    overflow: 'hidden',
    textOverflow: 'ellipsis',
    whiteSpace: 'nowrap',
    [theme.breakpoints.between('md', 'lg')]: {
      fontSize: 8.5,
    },
  },
  dateInfoWrapper: ({ chipWidth }: { chipWidth?: number }) => ({
    width: `calc(100% - ${chipWidth || CHIP_LARGE_DEFAULT_WIDTH}px)`,
    [theme.breakpoints.only('md')]: {
      width: `calc(100% - ${chipWidth || CHIP_SMALL_DEFAULT_WIDTH}px)`,
    },
    display: 'flex',
    justifyContent: 'flex-start',
  }),
  dateInfo: {
    width: `calc(100% - ${UPDATE_ICON_RIGHT_MARGIN + UPDATE_ICON_SIZE}px)`,
  },
}));

interface UpdateInfoProps {
  relativeTime?: string;
  timeText: string;
  image?: string;
  imageText?: string;
  live?: boolean;
  frequency?: 'hourly' | 'daily' | 'every 6 hours';
  subtitle?: string;
  href?: string;
  withMargin?: boolean;
  chipWidth?: number;
  onClick?: () => void;
}

export default UpdateInfo;
