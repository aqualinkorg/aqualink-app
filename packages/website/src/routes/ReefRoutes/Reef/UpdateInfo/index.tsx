import React from "react";
import {
  withStyles,
  WithStyles,
  createStyles,
  Grid,
  Box,
  Typography,
} from "@material-ui/core";
import UpdateIcon from "@material-ui/icons/Update";

const UpdateInfo = ({ timestamp, image, classes }: UpdateInfoProps) => (
  <Grid
    className={classes.updateInfo}
    container
    justify="space-around"
    alignItems="center"
    item
    spacing={1}
  >
    <Grid item>
      <Grid container alignItems="center" justify="center">
        <Grid item>
          <UpdateIcon className={classes.updateIcon} fontSize="small" />
        </Grid>
        <Grid item>
          <Box display="flex" flexDirection="column">
            <Typography variant="caption">
              Last data received {timestamp}
            </Typography>
            <Typography variant="caption">Updated daily</Typography>
          </Box>
        </Grid>
      </Grid>
    </Grid>
    <Grid className={classes.nooaChip} item>
      <Grid container alignItems="center">
        <Typography variant="caption">NOAA</Typography>
        <img className={classes.sensorImage} alt="sensor-type" src={image} />
      </Grid>
    </Grid>
  </Grid>
);

const styles = () =>
  createStyles({
    updateInfo: {
      backgroundColor: "#c4c4c4",
      color: "#757575",
      padding: " 0.5rem 0.125rem",
    },
    updateIcon: {
      marginRight: 5,
      height: "2rem",
      width: "2rem",
    },
    nooaChip: {
      backgroundColor: "#dddddd",
      borderRadius: 4,
      height: 26,
    },
    sensorImage: {
      height: 20,
      width: 20,
    },
  });

interface UpdateInfoIncomingProps {
  timestamp: string | null;
  image: string;
}

type UpdateInfoProps = UpdateInfoIncomingProps & WithStyles<typeof styles>;

export default withStyles(styles)(UpdateInfo);
