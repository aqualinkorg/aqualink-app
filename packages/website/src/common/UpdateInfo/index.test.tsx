import React from "react";
import { render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";

import UpdateInfo from ".";

test("renders as expected", () => {
  const { container } = render(
    <Router>
      <UpdateInfo
        relativeTime="02/01 14:00 EET"
        timeText="Last data received"
        image={null}
        imageText="NOAA"
        live={false}
        frequency="daily"
      />
    </Router>
  );
  expect(container).toMatchSnapshot();
});
