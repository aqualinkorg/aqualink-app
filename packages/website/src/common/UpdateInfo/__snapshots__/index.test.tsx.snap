// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as expected 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-item makeStyles-updateInfo-1 false css-1urtxze-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item makeStyles-dateInfoWrapper-5 makeStyles-dateInfoWrapper-7 css-x7bhrq-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container css-6zo1m7-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item css-x7bhrq-MuiGrid-root"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall makeStyles-updateIcon-3 css-120dh41-MuiSvgIcon-root"
            data-testid="UpdateIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M21 10.12h-6.78l2.74-2.82c-2.73-2.7-7.15-2.8-9.88-.1-2.73 2.71-2.73 7.08 0 9.79s7.15 2.71 9.88 0C18.32 15.65 19 14.08 19 12.1h2c0 1.98-.88 4.55-2.64 6.29-3.51 3.48-9.21 3.48-12.72 0-3.5-3.47-3.53-9.11-.02-12.58s9.14-3.47 12.65 0L21 3zM12.5 8v4.25l3.5 2.08-.72 1.21L11 13V8z"
            />
          </svg>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item makeStyles-dateInfo-6 css-x7bhrq-MuiGrid-root"
        >
          <mock-box
            display="flex"
            flexdirection="column"
            width="100%"
          >
            <mock-typography
              classname="makeStyles-updateInfoText-4"
              variant="caption"
            >
              Last data received 02/01 14:00 EET
            </mock-typography>
            <mock-typography
              classname="makeStyles-updateInfoText-4"
              variant="caption"
            >
              Updated daily
            </mock-typography>
          </mock-box>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item css-x7bhrq-MuiGrid-root"
      style="display: flex; justify-content: flex-end;"
    >
      <div
        class="MuiGrid-root MuiGrid-item makeStyles-chip-8 makeStyles-chip-14 css-x7bhrq-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-container css-6zo1m7-MuiGrid-root"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary makeStyles-button-13 css-iorydh-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            <mock-typography
              classname="makeStyles-chipText-9"
            >
              NOAA
            </mock-typography>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
